codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java

```diff
@@ -315,6 +321,61 @@ public static void setTransferEncodingChunked(HttpMessage m, boolean chunked) {
         }
     }
 
+    /**
+     * Fetch charset from message's Content-Type header.
+     *
+     * @return the charset from message's Content-Type header or {@link io.netty.util.CharsetUtil#ISO_8859_1}
+     * if charset is not presented or unparsable
+     */
+    public static Charset getCharset(HttpMessage message) {
+        return getCharset(message, CharsetUtil.ISO_8859_1);
+    }
+
+    /**
+     * Fetch charset from message's Content-Type header.
+     *
+     * @return the charset from message's Content-Type header or {@code defaultCharset}
+     * if charset is not presented or unparsable
+     */
+    public static Charset getCharset(HttpMessage message, Charset defaultCharset) {
+        CharSequence charsetCharSequence = getCharsetAsString(message);
+        if (charsetCharSequence != null) {
+            try {
+                return Charset.forName(charsetCharSequence.toString());
+            } catch (UnsupportedCharsetException unsupportedException) {
+                return defaultCharset;
+            }
+        } else {
+            return defaultCharset;
+        }
+    }
+
+    /**
+     * Fetch charset string from message's Content-Type header.
+     *
+     * A lot of sites/possibly clients have charset="CHARSET", for example charset="utf-8". Or "utf8" instead of "utf-8"
+     * This is not according to standard, but this method provide an ability to catch desired mistakes manually in code
+     *
+     * @return the charset string from message's Content-Type header or {@code null} if charset is not presented
+     */
+    public static CharSequence getCharsetAsString(HttpMessage message) {
+        CharSequence contentTypeValue = message.headers().get(HttpHeaderNames.CONTENT_TYPE);
+        if (contentTypeValue == null) {
+            return null;
+        }
+        int indexOfCharset = AsciiString.indexOfIgnoreCase(contentTypeValue, CHARSET_EQUALS, 0);
+        if (indexOfCharset != AsciiString.INDEX_NOT_FOUND) {
+            int indexOfEncoding = indexOfCharset + CHARSET_EQUALS.length();
+            if (indexOfEncoding < contentTypeValue.length()) {
+                return contentTypeValue.subSequence(indexOfEncoding, contentTypeValue.length());
+            } else {
+                return null;
+            }
+        } else {
```

Comment:
can you remove this `else` block and just do `return null;`?  Consider removing the previous `else` block too and just fall through to the `return null;` at the end of the method.